src/flow/src/adapter.rs

```diff
@@ -14,19 +14,678 @@
+                .assign_global_id_to_table(&self.table_info_source, None, Some(*source))
+                .await?;
+        }
+        node_ctx
+            .assign_global_id_to_table(&self.table_info_source, Some(sink_table_name.clone()), None)
+            .await?;
+
+        node_ctx.register_task_src_sink(flow_id, source_table_ids, sink_table_name.clone());
+
+        node_ctx.query_context = query_ctx.map(Arc::new);
+        // construct a active dataflow state with it
+        let flow_plan = sql_to_flow_plan(&mut node_ctx, &self.query_engine, &sql).await?;
+        debug!("Flow {:?}'s Plan is {:?}", flow_id, flow_plan);
+        node_ctx.assign_table_schema(&sink_table_name, flow_plan.typ.clone())?;
+
+        let expire_when = expire_when
+            .and_then(|s| {
+                if s.is_empty() || s.split_whitespace().join("").is_empty() {
+                    None
+                } else {
+                    Some(s)
+                }
+            })
+            .map(|d| {
+                let d = d.as_ref();
+                parse_fixed(d)
+                    .map(|(_, n)| n)
+                    .map_err(|err| err.to_string())
+            })
+            .transpose()
+            .map_err(|err| UnexpectedSnafu { reason: err }.build())?;
+        let _ = comment;
+        let _ = flow_options;
+
+        // TODO(discord9): add more than one handles
+        let sink_id = node_ctx.table_repr.get_by_name(&sink_table_name).unwrap().1;
+        let sink_sender = node_ctx.get_sink_by_global_id(&sink_id)?;
+
+        let source_ids = source_table_ids
+            .iter()
+            .map(|id| node_ctx.table_repr.get_by_table_id(id).unwrap().1)
+            .collect_vec();
+        let source_receivers = source_ids
+            .iter()
+            .map(|id| {
+                node_ctx
+                    .get_source_by_global_id(id)
+                    .map(|s| s.get_receiver())
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        let err_collector = ErrCollector::default();
+        self.flow_err_collectors
+            .write()
+            .await
+            .insert(flow_id, err_collector.clone());
+        let handle = &self.worker_handles[0].lock().await;
+        let create_request = worker::Request::Create {
+            flow_id,
+            plan: flow_plan,
+            sink_id,
+            sink_sender,
+            source_ids,
+            src_recvs: source_receivers,
+            expire_when,
+            create_if_not_exist,
+            err_collector,
+        };
+        handle.create_flow(create_request).await?;
+        info!("Successfully create flow with id={}", flow_id);
+        Ok(Some(flow_id))
+    }
+}
+
+/// FlowTickManager is a manager for flow tick, which trakc flow execution progress
+///
+/// TODO(discord9): better way to do it, and not expose flow tick even to other flow to avoid
+/// TSO coord mess
+#[derive(Clone)]
+pub struct FlowTickManager {
+    anchor: Anchor,
```

Comment:
`minstant::Anchor` is not suitable for long-term timing, it is recommended to use SystemTime instead